Maybe tolerance error for test_zero_constrained, macOS

On wheel testing:

```
=================================== FAILURES ===================================
______________ TestGenericNegativeBinomial.test_zero_constrained _______________
../venv/lib/python2.7/site-packages/statsmodels/base/tests/test_generic_methods.py:220: in test_zero_constrained
    assert_allclose(res1.resid, res2.resid, rtol=1e-10)
E   AssertionError: 
E   Not equal to tolerance rtol=1e-10, atol=0
E   
E   (mismatch 0.302129767211%)
E    x: array([-3.174188, -1.174188, -3.174188, ...,  5.713472,  5.686109,
E           3.730759])
E    y: array([-3.174188, -1.174188, -3.174188, ...,  5.713472,  5.686109,
E           3.730759])
```

Example:

https://travis-ci.org/MacPython/statsmodels-wheels/jobs/426380464

Build matrix:

https://travis-ci.org/MacPython/statsmodels-wheels/builds/426380455

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -10,12 +10,10 @@ jobs:
     name: Linux
     vmImage: 'ubuntu-16.04'
 
-# TODO: Enable these; they are currently disabled because
-#  test_lme.TestMixedLM.test_compare_numdiff fails on OSX
-# - template: tools/ci/azure_template.yml
-#  parameters:
-#    name: macOS
-#    vmImage: 'macOS-10.13'
+- template: tools/ci/azure_template.yml
+  parameters:
+    name: macOS
+    vmImage: 'macOS-10.13'
 
 - template: tools/ci/azure_template.yml
   parameters:

--- a/statsmodels/base/tests/test_generic_methods.py
+++ b/statsmodels/base/tests/test_generic_methods.py
@@ -12,6 +12,7 @@ Author: Josef Perktold
 """
 from statsmodels.compat.pandas import assert_series_equal, assert_index_equal
 from statsmodels.compat.python import range
+from statsmodels.compat.platform import PLATFORM_OSX
 
 import numpy as np
 import pandas as pd
@@ -112,7 +113,13 @@ class CheckGenericMixin(object):
         if hasattr(res1, 'resid'):
             # discrete models, Logit don't have `resid` yet
             # atol discussion at gh-5158
-            assert_allclose(res1.resid, res2.resid, rtol=1e-10, atol=1e-12)
+            rtol = 1e-10
+            atol = 1e-12
+            if PLATFORM_OSX:
+                # GH 5628
+                rtol = 1e-8
+                atol = 1e-10
+            assert_allclose(res1.resid, res2.resid, rtol=rtol, atol=atol)
 
         ex = self.results.model.exog.mean(0)
         predicted1 = res1.predict(ex, **self.predict_kwds)

--- a/statsmodels/compat/platform.py
+++ b/statsmodels/compat/platform.py
@@ -1,0 +1,5 @@
+import sys
+
+__all__ = ['PLATFORM_OSX']
+
+PLATFORM_OSX = sys.platform == 'darwin'

--- a/statsmodels/regression/tests/test_lme.py
+++ b/statsmodels/regression/tests/test_lme.py
@@ -1,3 +1,5 @@
+from statsmodels.compat.platform import PLATFORM_OSX
+
 import os
 import csv
 import warnings
@@ -86,6 +88,8 @@ def loglike_function(model, profile_fe, has_fe):
 class TestMixedLM(object):
 
     # Test analytic scores and Hessian using numeric differentiation
+    @pytest.mark.skipif(PLATFORM_OSX, reason='known to fail on OSX due to '
+                                             'numerical differences')
     @pytest.mark.slow
     @pytest.mark.parametrize("use_sqrt", [False, True])
     @pytest.mark.parametrize("reml", [False, True])

